github/Team.py

```diff
@@ -249,6 +268,26 @@ def set_repo_permission(self, repo, permission):
             "PUT", self.url + "/repos/" + repo._identity, input=put_parameters
         )
 
+    def update_team_repository(self, repo, permission):
+        """
+        :calls: `PUT /orgs/:org/teams/:team_slug/repos/:owner/:repo <https://developer.github.com/v3/teams/#add-or-update-team-repository>`_
+        :param repo: :class:`github.Repository.Repository`
+        :param permission: string
+        :rtype: None
+        """
+
+        # TODO Add support for Content-Length: 0 when no parameters are added
+        # Original note from the docs:
+        # Note that, if you choose not to pass any parameters, you'll need to set Content-Length to zero when calling out to this endpoint. For more information, see "HTTP verbs."
+
+        assert isinstance(repo, github.Repository.Repository), repo
+        put_parameters = {
+            "permission": permission,
+        }
+        headers, data = self._requester.requestJsonAndCheck(
+            "PUT", self.organization_url + "/teams/" + str(self.id)  + "/repos/" + repo._identity, input=put_parameters
+        )
+
```

Comment:
Do we want to return a new object here? I guess it's github.Permission.Permission.

Reply:
The official API returns only a `Status: 204 No Content`.
Maybe it would be good to return a bool if this was successful or not? What doy ou think?

Reply:
Ah, good point. Have a look at Repository.py to get a hint about we use that pattern.

Reply:
Fixed in https://github.com/PyGithub/PyGithub/pull/1509/commits/1e728aa738d97e075d8f0347e25931569e62fc6c